crates/store/src/db/mod.rs

```diff
@@ -254,6 +262,53 @@ impl Db {
         db_tx.commit().context("failed to commit database transaction")
     }
 
+    /// Avoid repeated boilerplate, frame the query in a transaction
+    pub(crate) async fn framed<R, E, Q, M>(&self, msg: M, query: Q) -> std::result::Result<R, E>
```

Comment:
Naming is hard but this did confuse me a bit. Is `framed` a known term for this?

Maybe `fn transact` or `fn query` or just `transaction`.

Reply:
It's not, it just put a "frame" around the call and `transaction` is just to omnipresent.

Reply:
Rename to `transact` and `query`